src/dune_rules/inline_tests.ml

```diff
@@ -183,6 +184,7 @@ include Sub_system.Register_end_point (struct
         (let+ loc = loc
          and+ deps = field "deps" (repeat Dep_conf.decode) ~default:[]
          and+ flags = Ordered_set_lang.Unexpanded.field "flags"
+         and+ compile_flags = Ocaml_flags.Spec.decode_compile_flags_field "compile_flags"
```

Comment:
This should be guarded behind a `2.8` version check.

Reply:
it's not clear what you mean here @rgrinberg.

Reply:
Allow me to clarify:

In dune, we may not introduce new features without guarding them behind the version of the dune language in which they were introduced. In this particular case, we are introducing a new `compile_flags` field in version 2.8 of dune. Hence, it should only be available once the user selected this version in the project file: `(lang dune 2.8)`.

So you need a check like this `field "compile_flag" (Dune_lang.Syntax.since syntax (2, 8) >>> ..)` when defining this field. See `dune_file.ml` for instances of such checks.

Reply:
Once you rebase `(2, 8)` should now be available and you'll be able to add the syntax check on `(2, 8)` above.